source/common/config/grpc_mux_subscription_impl.h

```diff
@@ -64,7 +64,7 @@ class GrpcMuxSubscriptionImpl : public Subscription<ResourceType>,
     // TODO(htuch): Less fragile signal that this is failure vs. reject.
     if (e == nullptr) {
       stats_.update_failure_.inc();
-      ENVOY_LOG(warn, "gRPC update for {} failed", type_url_);
+      ENVOY_LOG(debug, "gRPC update for {} failed", type_url_);
```

Comment:
Can you distinguish between the very first update and alter updates? In particular, I think we should warn when a management server is unreachable etc. but not during normal disconnects (e.g. gRPC load shed).

Reply:
If it is not unreachable during Envoy startup, we get the following warning already
[2018-06-27 13:47:24.884][27][warning][upstream] source/common/config/grpc_mux_impl.cc:235] gRPC config stream closed: 14, upstream connect error or disconnect/reset before headers
 
Is this sufficient? or are you thinking of some thing else?

Reply:
Ok. So do you want this to be warn till at least one successful update has happened or one update attempt? Based on that, I think we can evaluate the corresponding stat and distinguish between debug and warn.

Reply:
OK, seems fine then if we are already observing a connection level warning.